app/core/models/abortables.ts

```diff
@@ -0,0 +1,54 @@
+import {nanoid} from "@reduxjs/toolkit"
+import {isString, remove} from "lodash"
+
+export type Abortable<Meta extends any = any> = {
+  id: string
+  tab?: string
+  tag?: string
+  meta?: Meta
+  abort: () => void | Promise<void>
```

Comment:
I like this idea. How about the individual abort function that gets added can return void or a promise, but when calling abort through the container class, it always returns a promise. For example:

```js
// This way we don't need to wrap this in a promise everywhere
abortables.add({id: "1", abort: () => ctl.abort})

// But this will always be a promise
const promise = abortables.abort("1")
```

Reply:
Yeah, that sounds great 👍